sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -238,28 +274,43 @@ internal static IEnumerable<HttpHeader> GetHeaders(HttpHeaders headers, HttpCont
                     yield return new HttpHeader(header.Key, JoinHeaderValues(header.Value));
                 }
             }
+#endif
+
         }
 
         internal static bool RemoveHeader(HttpHeaders headers, HttpContent? content, string name)
         {
             // .Remove throws on invalid header name so use TryGet here to check
+#if NET6_0_OR_GREATER
+            if (headers.NonValidated.Contains(name) && headers.Remove(name))
+            {
+                return true;
+            }
+
+            return content is not null && content.Headers.NonValidated.Contains(name) && content.Headers.Remove(name);
```

Comment:
Is the `Contains` call necessary here, or is it faster to check existence first in NonValidated?
```suggestion
            return headers.Remove(name) || content is not null && content.Headers.Remove(name);
```

Reply:
It depends on what we do more often - remove existing or non-existing header. If header is added, then yes - simple call of `headers.Remove` is faster. And header is missing, then `headers.NonValidated.Contains` validates faster than `headers.Remove`. Probably we can use your approach.

Reply:
It looks like `headers.Remove` still thrown on invalid header, so we must call  `headers.NonValidated.Contains` first